concepts/objects/about.md

```diff
@@ -0,0 +1,251 @@
+There you can omit the colon and the `function` keyword.
+
+```javascript
+const x = 1;
+const y = 2;
+
+const obj = {
+  x: x,
+  y: y,
+  calcSum: function (a, b) {
+    return a + b;
+  },
+};
+
+// Can be shortened to ...
+const obj = {
+  x,
+  y,
+  calcSum(a, b) {
+    return a + b;
+  },
+};
+```
+
+If you want to create an object literal with a key name that is stored in a variable or derived via an expression, you need to wrap the variable or expression in square brackets.
+
+```javascript
+const key = 'greeting';
+
+const obj = {
+  [key]: 'hello world',
+};
+```
+
+## Retrieving a Value
+
+There are two ways to retrieve the value for a given key, dot notation and bracket notation.
+
+```javascript
+const obj = { greeting: 'hello world' };
+
+obj.greeting;
+// => hello world
+
+obj['greeting'];
+// => hello world
+
+// Bracket notation also works with variables.
+const key = 'greeting';
+obj[key];
+// => hello world
+```
+
+Using the dot notation as a short-hand has the same restriction as omitting the quotation marks.
+It only works if the key follows the identifier naming rules.
+
+If you try to retrieve a key that does not exist in the object, JavaScript returns `undefined`.
+See the [Null and Undefined Concept][concept-null-undefined] for more details on this.
+
+You can chain the keys if you want to retrieve a value from a nested object.
+
+```javascript
+const obj = {
+  address: {
+    street: 'Trincomalee Highway',
+    city: 'Batticaloa',
+  },
+};
+
+obj.address.city;
+// => 'Batticaloa'
+
+obj['address']['city'];
+// => 'Batticaloa'
+```
+
+## Adding or Changing a Value
+
+You can add or change a value using the assignment operator `=`.
+Again, there is dot and bracket notation available.
+
+```javascript
+const obj = { greeting: 'hello world' };
+
+obj.greeting = 'Hi there!';
+obj['greeting'] = 'Welcome.';
+
+obj.newKey1 = 'new value 1';
+obj['new key 2'] = 'new value 2';
+
+const key = 'new key 3';
+obj[key] = 'new value 3';
+```
+
+## Deleting an Entry
+
+You can delete a key-value pair from an object using the `delete` keyword.
+
+```javascript
+const obj = {
+  key1: 'value1',
+  key2: 'value2',
+};
+
+delete obj.key1;
+delete obj['key2'];
+```
+
+Note that although `undefined` is returned for non-existing keys, setting a key to `undefined` does not remove the entry.
+Always use `delete` instead.
+
+## Checking Whether a Key Exists
+
+You can check whether a certain key exists in an object with the `hasOwnProperty` method.
+
+```javascript
+const obj = { greeting: 'hello world' };
+
+obj.hasOwnProperty('greeting');
+// => true
+
+obj.hasOwnProperty('age');
+// => false
+```
+
+`hasOwnProperty` returns `false` for [inherited keys][concept-inheritance]. This is usually the desired behavior.
+If you want include inherited keys in the existence check you can use the [`in` operator][mdn-in-operator] instead.
+
+## Looping Through an Object
+
+There is a special `for...in` loop to iterate over all keys of an object.
```

Comment:
Are objects in JS iterables? Would `for…of` work too?

Reply:
`for...in` and `for...of` are not equivalent, and the concept of enumerable is not the same as iterable, in JavaScript (@junedev all this might be worth mentioning, somewhere).

JavaScript objects have _enumerable properties_. They are enumerable by default, and can be made non enumerable using `Object.create` or `Object.defineProperty` setting `{ enumerable: false }`.

They are _only_ iterable if there is `Symbol.iterator` defined (as a method):

```javascript
const iterable1 = {};

iterable1[Symbol.iterator] = function* () {
  yield 1;
  yield 2;
  yield 3;
};

console.log([...iterable1]);
// [1, 2, 3]
```

[Array's prototype has it defined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/@@iterator), but Object's prototype does not. _Iterability_ is **only** used by `for ... of`. The async variant is `Symbol.asyncIterator` and is used by **for-await-of**.

Reply:
@SleeplessByte I would not go into the details of this in this concept. Left a note (and the link to this comment) on our concept tracking issue. https://github.com/exercism/javascript/issues/951#issuecomment-864452573

Reply:
Depending on what comes first in the track, this may be worth mentioning, i.e. if `for..of` is explained in a previous exercise. That way, the student knows to watch out that `for..of` won't work here.

Thanks for the explanation :)

Reply:
`for...of` comes later in our current unlock flow but we treat about.md as independent from that flow. So your point is totally valid. I added a small part on this. I used a link to the MDN docs for now. Once we have our own concept on enumerable and iterable we can link that instead.